test/fail_compilation/template_decl.d

```diff
@@ -0,0 +1,9 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/template_decl.d(7): Error: `{` expected after template parameter list, not `(`
+fail_compilation/template_decl.d(7): Error: declaration expected, not `(`
```

Comment:
I don't think this has to be fixed in this PR, but the second error is redundant.

Reply:
Yes, not sure how to fix. There are also redundant errors if there's no identifier after the template keyword.